docs.feldera.com/docs/tutorials/api-client.md

```diff
@@ -0,0 +1,93 @@
+# REST API client
+
+## Request handling
+
+### Request outcome expectations
+
+The outcome of a request is that it either fails (e.g., DNS lookup failed) without any response
+(no status code nor body), or it succeeds and gets back a response status code and body.
+
+In case of a response, usually it is the Feldera implementation that generated it:
+- If it is success (2xx), it will return whichever body belongs to the success response.
+- Otherwise, if it is an error (4xx, 5xx), it will return a Feldera error response JSON body
+  which will have an application-level `error_code`.
+
+However, there are two notable exceptions in which case the response is not generated by
```

Comment:
would "in which case" be better as "when"?

Reply:
Simplified to "when"